scripts-available/CDB_FederatedServerTables.sql

```diff
@@ -0,0 +1,345 @@
+--
+-- Postcondition: it generates a view in the schema of the user that
+-- can be used through SQL and Maps API's.
+-- If the table was already exported, it will be dropped and re-imported
+--
+-- The view is placed under the server's view schema (cdb_fs_$(server_name))
+--
+-- E.g:
+-- SELECT cartodb.CDB_SetUp_PG_Federated_Table(
+--   'amazon',                  -- mandatory, name of the federated server
+--   'my_remote_schema',        -- mandatory, schema name
+--   'my_remote_table',         -- mandatory, table name
+--   'id',                      -- mandatory, name of the id column
+--   'geom',                    -- optional, name of the geom column, preferably in 4326
+--   'webmercator'              -- optional, should be in 3857 if present
+--   'local_name'               -- optional, name of the local view (uses the remote_name if not declared)
+-- );
+--
+CREATE OR REPLACE FUNCTION @extschema@.CDB_Federated_Table_Register(
+    server TEXT,
+    remote_schema TEXT,
+    remote_table TEXT,
+    id_column TEXT,
+    geom_column TEXT DEFAULT NULL,
+    webmercator_column TEXT DEFAULT NULL,
+    local_name NAME DEFAULT NULL
+)
+RETURNS void
+AS $$
+DECLARE
+    server_internal name := @extschema@.__CDB_FS_Generate_Server_Name(input_name => server, check_existence => false);
+    local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, remote_schema);
+    role_name name := @extschema@.__CDB_FS_Generate_Server_Role_Name(server_internal);
+
+    src_table REGCLASS; -- import_schema.remote_table  - import_schema is local_schema
+    local_view text;    -- view_schema.local_name      - view_schema is server_internal
+
+    rest_of_cols TEXT[];
+    geom_expression TEXT;
+    webmercator_expression TEXT;
+    carto_columns_expression TEXT[];
+BEGIN
+    IF remote_table IS NULL THEN
+        RAISE EXCEPTION 'Remote table name cannot be NULL';
+    END IF;
+
+    -- Make do with whatever columns are provided
+    IF webmercator_column IS NULL THEN
+        webmercator_column := geom_column;
+    ELSIF geom_column IS NULL THEN
+        geom_column := webmercator_column;
+    END IF;
+
+    IF local_name IS NULL THEN
+        local_name := remote_table;
+    END IF;
+
+    -- Import the foreign table
+    -- Drop the old view / table if there was one
+    IF EXISTS (SELECT 1 FROM information_schema.tables WHERE table_schema = local_schema AND table_name = remote_table) THEN
+        EXECUTE @extschema@.CDB_Federated_Table_Unregister(server, remote_schema, remote_table);
+    END IF;
+    BEGIN
+        EXECUTE FORMAT('IMPORT FOREIGN SCHEMA %I LIMIT TO (%I) FROM SERVER %I INTO %I;',
+                        remote_schema, remote_table, server_internal, local_schema);
+    EXCEPTION WHEN OTHERS THEN
+        RAISE EXCEPTION 'Could not import schema "%" of server "%": %', remote_schema, server, SQLERRM;
+    END;
+    
+    BEGIN
+        src_table := format('%I.%I', local_schema, remote_table);
+    EXCEPTION WHEN OTHERS THEN
```

Comment:
weird place for an exception to happen (guess this happened during testing)

Reply:
Yes, casting `'%I.%I'` to REGCLASS will raise an exception if the object doesn't exist.